src/core/abr/abr_manager.ts

```diff
@@ -148,3 +151,38 @@ export default class ABRManager {
     return originalBandwidthEstimator;
   }
 }
+
+/**
+ * Create Observable that merge several throttling Observables into one.
+ * @param {Observable} limitWidth$ - Emit the width at which the chosen
+ * Representation should be limited.
+ * @param {Observable} throttleBitrate$ - Emit the maximum bitrate authorized.
+ * @param {Observable} throttle$ - Also emit the maximum bitrate authorized.
+ * Here for legacy reasons.
+ * @returns {Observable}
+ */
+function createFilters(
+  limitWidth$? : Observable<number>,
+  throttleBitrate$? : Observable<number>,
+  throttle$? : Observable<number>
+) : Observable<IABRFiltersObject> {
+  const deviceEventsArray : Array<Observable<IABRFiltersObject>> = [];
+
+  if (limitWidth$ != null) {
+    deviceEventsArray.push(limitWidth$.pipe(map(width => ({ width }))));
+  }
+  if (throttle$ != null) {
+    deviceEventsArray.push(throttle$.pipe(map(bitrate => ({ bitrate }))));
+  }
+  if (throttleBitrate$ != null) {
+    deviceEventsArray.push(throttleBitrate$.pipe(map(bitrate => ({ bitrate }))));
+  }
+
+  // Emit restrictions on the pools of available representations to choose
+  // from.
+  return deviceEventsArray.length > 0 ?
+    observableCombineLatest(deviceEventsArray)
+      .pipe(map((args : IABRFiltersObject[]) : IABRFiltersObject =>
+        objectAssign({}, ...args) as IABRFiltersObject)) :
```

Comment:
Why the `as` is necessary?

also does breaking the line could improve readability here ?

```
 return deviceEventsArray.length > 0 
   ? observableCombineLatest(deviceEventsArray)
      .pipe(
        map((args : IABRFiltersObject[]) : IABRFiltersObject =>objectAssign({}, ...args) as IABRFiltersObject)
       )
   : observableOf({});
```

Reply:
> Why the as is necessary?

I don't remember, but it may be because we're using a rest parameter which are harder to type. Maybe TS does not recognize the type? I don't remember.

Also I have more trouble following the one-liner personally, due to the amount of type information everywhere

Reply:
So I re-check. Is because Object.assign is typed as such with rest parameters:
```ts
assign(target: object, ...sources: any[]): any;
```

This in turn lead our linter to throw due to the `no-unsafe-return` rule.